data.go

```diff
@@ -85,86 +64,289 @@ func setDB(db apid.DB) {
 	dbMux.Unlock()
 }
 
-func insertApidCluster(dac dataApidCluster, txn *sql.Tx) error {
+//TODO if len(rows) > 1000, chunk it up and exec multiple inserts in the txn
+func insert(tableName string, rows []common.Row, txn *sql.Tx) bool {
 
-	log.Debugf("inserting into APID_CLUSTER: %v", dac)
+	if len(rows) == 0 {
+		return false
+	}
 
-	//replace to accomodate same snapshot txid
-	stmt, err := txn.Prepare(`
-	REPLACE INTO APID_CLUSTER
-		(id, description, name, umbrella_org_app_name,
-		created, created_by, updated, updated_by,
-		last_sequence)
-	VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9);
-	`)
-	if err != nil {
-		log.Errorf("prepare insert into APID_CLUSTER transaction Failed: %v", err)
-		return err
+	var orderedColumns []string
+	for column := range rows[0] {
+		orderedColumns = append(orderedColumns, column)
 	}
-	defer stmt.Close()
+	sort.Strings(orderedColumns)
 
-	_, err = stmt.Exec(
-		dac.ID, dac.Description, dac.Name, dac.OrgAppName,
-		dac.Created, dac.CreatedBy, dac.Updated, dac.UpdatedBy,
-		"")
+	sql := buildInsertSql(tableName, orderedColumns, rows)
 
+	prep, err := txn.Prepare(sql)
 	if err != nil {
-		log.Errorf("insert APID_CLUSTER failed: %v", err)
+		log.Errorf("INSERT Fail to prepare statement [%s] error=[%v]", sql, err)
+		return false
 	}
+	defer prep.Close()
 
-	return err
-}
+	var values []interface{}
 
-func insertDataScope(ds dataDataScope, txn *sql.Tx) error {
+	for _, row := range rows {
+		for _, columnName := range orderedColumns {
+			//use Value so that stmt exec does not complain about common.ColumnVal being a struct
+			//TODO will need to convert the Value (which is a string) to the appropriate field, using type for mapping
```

Comment:
yes, I will delete that comment.  The go sql driver does automatic type conversions.  What we will need to do eventually is fix the way transicator chooses to map PG types to sqlite types.  For instance, date columns in PG are mapped to blobs in sqlite, and dates are inserted as strings.